libraries/AP_IBus_Telem/AP_IBus_Telem.h

```diff
@@ -0,0 +1,133 @@
+#define IBUS_MEAS_TYPE_CMP_HEAD       0x08 // Heading
+#define IBUS_MEAS_TYPE_CLIMB_RATE     0x09 // Climb rate
+#define IBUS_MEAS_TYPE_COG            0x0a // Course over ground
+#define IBUS_MEAS_TYPE_GPS_STATUS     0x0b // GPS status (2 values)
+#define IBUS_MEAS_TYPE_ACC_X          0x0c // Acc X
+#define IBUS_MEAS_TYPE_ACC_Y          0x0d // Acc Y
+#define IBUS_MEAS_TYPE_ACC_Z          0x0e // Acc Z
+#define IBUS_MEAS_TYPE_ROLL           0x0f // Roll
+#define IBUS_MEAS_TYPE_PITCH          0x10 // Pitch
+#define IBUS_MEAS_TYPE_YAW            0x11 // Yaw
+#define IBUS_MEAS_TYPE_VERTICAL_SPEED 0x12 // Vertical speed
+#define IBUS_MEAS_TYPE_GROUND_SPEED   0x13 // Speed m/s
+#define IBUS_MEAS_TYPE_GPS_DIST       0x14 // Distance from home
+#define IBUS_MEAS_TYPE_ARMED          0x15 // Armed / unarmed (1 = armed, 0 = unarmed)
+#define IBUS_MEAS_TYPE_FLIGHT_MODE    0x16 // Flight mode (see Mode.h for flightmodes in copter)
+
+#define IBUS_MEAS_TYPE_PRES           0x41 // Pressure
+#define IBUS_MEAS_TYPE_SPE            0x7e // Speed km/h
+
+class AP_IBus_Telem
+{
+public:
+    //constructor
+    AP_IBus_Telem() {}
+
+    /* Do not allow copies */
+    AP_IBus_Telem(const AP_IBus_Telem &other) = delete;
+    AP_IBus_Telem &operator=(const AP_IBus_Telem &) = delete;
+
+    void init();
+    void loop();
+
+private:
+    AP_HAL::UARTDriver *_port;
+    bool _initialized;
+
+    uint8_t addSensor(uint8_t type, uint8_t len = 2); // add sensor type and data length (2 or 4), returns address
+    void addSensors(); //add sensors for transmitting via IBus
+    void setSensorMeasurement(uint8_t adr, int32_t value); //sets a sensor measurement for a certain address
+    void setSensorMeasurements(); //set new values for all sensors configured
+    void populate_checksum(uint8_t *packet, uint16_t size); //populates the last two bytes with checksum
+
+    enum State {
+        GET_LENGTH,
+        GET_DATA,
+        GET_CHKSUML,
+        GET_CHKSUMH,
+        DISCARD
+    };
+
+    static const uint8_t PROTOCOL_MAX_LENGTH = 0x08;           //maximum length for sensor frames - 8 bytes
+    static const uint8_t PROTOCOL_OVERHEAD = 3;            // packet is <len><cmd><data....><chkl><chkh>, overhead=cmd+chk bytes
+    static const uint8_t PROTOCOL_COMMAND_DISCOVER = 0x80; // Command discover sensor (lowest 4 bits are sensor)
+    static const uint8_t PROTOCOL_COMMAND_TYPE = 0x90;     // Command discover sensor (lowest 4 bits are sensor)
+    static const uint8_t PROTOCOL_COMMAND_VALUE = 0xA0;    // Command send sensor data (lowest 4 bits are sensor)
+    static const uint8_t PROTOCOL_TIMEGAP = 3; // Packets are received very ~7ms so use ~half that for the gap
+    static const uint16_t UPDATE_MEASUREMENTS_TIMING = 500; //every 500ms
+    static const uint8_t PROTOCOL_FOUR_LENGTH = 0x04;      //indicate that the message has 4 bytes
+    static const uint8_t PROTOCOL_SIX_LENGTH = 0x06;      //indicate that the message has 6 bytes
+
+    State _state;
```

Comment:
```suggestion
    State _state = State::GET_LENGTH;
```

That is the same value as it will currently get, but particularly for enumerated values its worth putting in.